### PR TITLE
[API]fixed behat step change-quantity with minor improvements

### PR DESCRIPTION
**Pull request by arti0090**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| License         | MIT

With new tests regarding changing order after completed checkout, one of endpoints i have double tested where I should use other endpoint - this is fixed here. Also I added some minor fixes to CheckoutContext file

### CODE REVIEW

File: features/checkout/preventing_cart_from_being_modified_after_checkout.feature [hunk 1]
```diff
@@ -63,9 +63,8 @@ Feature: Preventing cart from being modified after checkout
     @api
     Scenario: Preventing from changing quantity of product after checkout
         Given I added product "Sig Sauer P226" to the cart
-        And I added product "AK-47" to the cart
         Then I specified the billing address as "Ankh Morpork", "Frost Alley", "90210", "United States" for "Jon Snow"
         And I proceeded with "Free" shipping method and "Cash on Delivery" payment
         And I confirmed my order
-        When I try to change quantity to 2 of product "AK-47" from the cart
+        When I try to change quantity of first product to 2 in cart
```
Comment (GSadee) [inline features/checkout/preventing_cart_from_being_modified_after_checkout.feature:69]:
```suggestion
        When I try to change quantity to 2 of product "Sig Sauer P226" from the cart
```
?

File: src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php [hunk 2]
```diff
@@ -827,8 +818,9 @@ private function hasShippingMethodWithFee(ShippingMethodInterface $shippingMetho
     {
         foreach ($this->getCartShippingMethods($this->getCart()) as $cartShippingMethod) {
             if (
-                $cartShippingMethod['shippingMethod']['code'] === $shippingMethod->getCode() &&
-                $cartShippingMethod['cost'] === $fee
+                $cartShippingMethod['cost'] === $fee &&
+                $cartShippingMethod['shippingMethod']['code'] === $shippingMethod->getCode()
+
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php:823]: 😱 
```suggestion
```

File: src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php [hunk 3]
```diff
@@ -941,6 +933,23 @@ private function putProductToCart(ProductInterface $product, string $tokenValue,
         );
     }
 
+    private function changeQuantityOfFirstProductInCart(string $tokenValue, int $quantity = 1): void
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php:936]: As we have almost 1000 lines in this context, these methods for operating on cart like adding products or changing quantity could be moved to the CartContext

**Review by AdamKasp — APPROVED** [review]

**Comment by arti0090** [comment]
After some bigger changes, this PR is hard to bump to newest master (conflicts). WIll be splitted to other PR's

**Pull request closed without merging**